src/qamqpclient.cpp

```diff
@@ -660,7 +660,11 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
-    QAmqpExchange *exchange = new QAmqpExchange(channelNumber, this);
+    QAmqpExchange *exchange = static_cast<QAmqpExchange*>(d->exchanges.get(name));
```

Comment:
the "Qt container" way of doing this would rather be to provide a "contains" method on QAmqpChannelHash:

```
if (d->exchanges.contains(name))
   return d->exchanges.get(name);

exchange = ...
```